### PR TITLE
Remove old solver interfaces on core classes #646

### PR DESCRIPTION
**Pull request by stairs**
Remove a few attributes from old solver interfaces and deprecated attributes in the solution model.

### CODE REVIEW

File: cobra/flux_analysis/loopless.py [hunk 1]
```diff
@@ -272,18 +272,15 @@ def construct_loopless_model(cobra_model):
             thermo_stoic["thermo_var_" + met.id][bound_id] = stoic
         # I * 1000 > v --> I * 1000 - v > 0
         reaction_ind = Reaction(reaction.id + "_indicator")
-        reaction_ind.variable_kind = "integer"
```
Comment (Midnighter) [inline cobra/flux_analysis/loopless.py]: Please remove the entire function `construct_loopless_model`and its reference in `cobra/flux_analysis/__init__.py`. Thanks!

**Comment by stairs** [comment]
@Midnighter Thanks a lot for the review! I'll add the changes during the weekend.

**Comment by stairs** [comment]
@Midnighter Looks like pytest was updated recently to 4.0.0 and all the RemovedInPytest4Warnings are now errors by default. I assume a PR should be merged first to fix all the tests.

**Comment by Midnighter** [comment]
Yes, you're correct. Thank you for making those adjustments. I will get the test suite up and running today.

**Comment by Midnighter** [comment]
I have created https://github.com/opencobra/cobrapy/pull/782 and it looks good so far. Just need to have it reviewed and then we can merge your work afterwards.

**Comment by Midnighter** [comment]
Okay, #782 has been merged. If you can rebase your branch on the latest devel, I'll be happy to merge it.

**Comment by Midnighter** [comment]
Actually, there were no conflicts so I just went ahead and merged it. Thank you for your work!

**Pull request merged**